python/lancedb/query.py

```diff
@@ -579,3 +620,169 @@ def to_arrow(self) -> pa.Table:
+        fts_results = self._normalize_scores(fts_results, "score")
+
+        results = self._reranker.rerank_hybrid(self, vector_results, fts_results)
+        if not isinstance(results, pa.Table):  # Enforce type
+            raise TypeError(
+                f"rerank_hybrid must return a pyarrow.Table, got {type(results)}"
+            )
+
+        if not self._with_row_id:
+            results = results.drop(["_rowid"])
+        return results
+
+    def _rank(self, results: pa.Table, column: str, ascending: bool = True):
+        if len(results) == 0:
+            return results
+        # Get the _score column from results
+        scores = results.column(column).to_numpy()
+        sort_indices = np.argsort(scores)
+        if not ascending:
+            sort_indices = sort_indices[::-1]
+        ranks = np.empty_like(sort_indices)
+        ranks[sort_indices] = np.arange(len(scores)) + 1
+        # replace the _score column with the ranks
+        _score_idx = results.column_names.index(column)
+        results = results.set_column(
+            _score_idx, column, pa.array(ranks, type=pa.float32())
+        )
+        return results
+
+    def _normalize_scores(self, results: pa.Table, column: str, invert=False):
+        if len(results) == 0:
+            return results
+        # Get the _score column from results
+        scores = results.column(column).to_numpy()
+        # normalize the scores by subtracting the min and dividing by the max
+        max, min = np.max(scores), np.min(scores)
+        if np.isclose(max, min):
+            rng = max
+        else:
+            rng = max - min
+        scores = (scores - min) / rng
+        if invert:
+            scores = 1 - scores
+        # replace the _score column with the ranks
+        _score_idx = results.column_names.index(column)
+        results = results.set_column(
+            _score_idx, column, pa.array(scores, type=pa.float32())
+        )
+        return results
+
+    def rerank(
+        self,
+        normalize="score",
+        reranker: Reranker = LinearCombinationReranker(weight=0.7, fill=1.0),
+    ) -> LanceHybridQueryBuilder:
+        """
+        Rerank the hybrid search results using the specified reranker. The reranker
+        must be an instance of Reranker class.
+
+        Parameters
+        ----------
+        normalize: str, default "score"
+            The method to normalize the scores. Can be "rank" or "score". If "rank",
+            the scores are converted to ranks and then normalized. If "score", the
+            scores are normalized directly.
+        reranker: Reranker, default LinearCombinationReranker(weight=0.7, fill=1.0)
+            The reranker to use. Must be an instance of Reranker class.
+        Returns
+        -------
+        LanceHybridQueryBuilder
+            The LanceHybridQueryBuilder object.
+        """
+        # TODO: Probably auto can be removed and score should be the default.
```

Comment:
Not clear what auto does here? If it's not used commonly then it's best to remove.

Reply:
oh I already removed it but not the docs